include/xtensor/xreducer.hpp

```diff
@@ -95,7 +97,13 @@ namespace xt
         using xexpression_type = std::decay_t<CT>;
         using axes_type = X;
 
-        using value_type = typename xexpression_type::value_type;
+        struct result_type_helper
+        {
+            functor_type m_f;
+        };
+
+        using substepper_type = typename xexpression_type::const_stepper;
+        using value_type = std::decay_t<decltype(((result_type_helper*)0)->m_f(**(substepper_type*)0, **(substepper_type*)0))>;
```

Comment:
Does the following work ?
```cpp
using value_type = std::decay_t<decltype(std::declval<functor_type>()(*std::declval<substepper_type>(), *std::declval<substepper_type>())>;
```
That would avoid the necessity for `result_type_helper`

Reply:
Indeed, thanks for pointing this out. I updated the PR accordingly.